src/main/java/org/apache/camel/idea/util/IdeaUtils.java

```diff
@@ -316,8 +318,13 @@ public static boolean isCamelExpressionOrLanguage(PsiClass clazz) {
     private static boolean isFromJavaMethod(PsiMethodCallExpression call, String... methods) {
         PsiMethod method = call.resolveMethod();
         if (method != null) {
-            String name = method.getName();
-            return Arrays.stream(methods).anyMatch(name::equals);
+            PsiClass containingClass = method.getContainingClass();
+            if (containingClass != null) {
+                String className = containingClass.getQualifiedName();
+                String name = method.getName();
+
+                return Arrays.stream(methods).anyMatch(name::equals) && "org.apache.camel.builder.RouteBuilder".equals(className);
```

Comment:
People can create base classes that extends RouteBuilder, so you would need to look the entire extends hierachy to find that class somewhere. I think PsiClass has some method to get all the extends and their prarents.